insta/app/controllers/users_controller.rb

```diff
@@ -0,0 +1,25 @@
+class UsersController < ApplicationController
+  before_action :authenticate_user!, only: [:edit, :update]
+
+  def show
+    @user  = User.find(params[:id])
+    @posts = current_user.posts.order(created_at: :desc)
+  end
+
+  def edit
+    @user = User.find(params[:id]) #finds the current user to edit that profile
+  end
+
+  def update
+    current_user.update(user_params) #update the current user based on params in form edit
+    redirect_to current_user         #goes to private method 
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:username, :name, :website, :bio, :email, :phone, :gender, :avatar) #avatar added
+  end
+  #Rails Strong Parameters: we permit these params.  If they aren't permitted, it will be blocked for security
+
+end
```

Comment:
Something doesn't quite work here (assuming it's to do with devise). When you sign out and go to sign_out, it's calling the show method which looks for a @user with an id "sign_out", and it's throwing an error because one isn't found.